device/blackhole_implementation.cpp

```diff
@@ -18,77 +18,43 @@ std::tuple<xy_pair, xy_pair> blackhole_implementation::multicast_workaround(xy_p
 }
 
 tlb_configuration blackhole_implementation::get_tlb_configuration(uint32_t tlb_index) const {
-    if (tlb_index >= blackhole::TLB_BASE_INDEX_16M) {
-        return tlb_configuration{
-            .size = blackhole::DYNAMIC_TLB_16M_SIZE,
-            .base = blackhole::DYNAMIC_TLB_16M_BASE,
-            .cfg_addr = blackhole::DYNAMIC_TLB_16M_CFG_ADDR,
-            .index_offset = tlb_index - blackhole::TLB_BASE_INDEX_16M,
-            .offset = blackhole::TLB_16M_OFFSET,
-        };
-    } else if (tlb_index >= blackhole::TLB_BASE_INDEX_2M) {
-        return tlb_configuration{
-            .size = blackhole::DYNAMIC_TLB_2M_SIZE,
-            .base = blackhole::DYNAMIC_TLB_2M_BASE,
-            .cfg_addr = blackhole::DYNAMIC_TLB_2M_CFG_ADDR,
-            .index_offset = tlb_index - blackhole::TLB_BASE_INDEX_2M,
-            .offset = blackhole::TLB_2M_OFFSET,
-        };
-    } else {
-        return tlb_configuration{
-            .size = blackhole::DYNAMIC_TLB_1M_SIZE,
-            .base = blackhole::DYNAMIC_TLB_1M_BASE,
-            .cfg_addr = blackhole::DYNAMIC_TLB_1M_CFG_ADDR,
-            .index_offset = tlb_index - blackhole::TLB_BASE_INDEX_1M,
-            .offset = blackhole::TLB_1M_OFFSET,
-        };
-    }
+    return tlb_configuration{
+        .size = blackhole::DYNAMIC_TLB_2M_SIZE,
+        .base = blackhole::DYNAMIC_TLB_2M_BASE,
+        .cfg_addr = blackhole::DYNAMIC_TLB_2M_CFG_ADDR,
+        .index_offset = tlb_index - blackhole::TLB_BASE_INDEX_2M,
+        .offset = blackhole::TLB_2M_OFFSET,
+    };
 }
 
 std::optional<std::tuple<std::uint32_t, std::uint32_t>> blackhole_implementation::describe_tlb(
     std::int32_t tlb_index) const {
-    std::uint32_t TLB_COUNT_1M = 156;
-    std::uint32_t TLB_COUNT_2M = 10;
-    std::uint32_t TLB_COUNT_16M = 20;
+    std::uint32_t TLB_COUNT_2M = 202;
 
-    std::uint32_t TLB_BASE_1M = 0;
-    std::uint32_t TLB_BASE_2M = TLB_COUNT_1M * (1 << 20);
-    std::uint32_t TLB_BASE_16M = TLB_BASE_2M + TLB_COUNT_2M * (1 << 21);
+    std::uint32_t TLB_BASE_2M = 0;
     if (tlb_index < 0) {
         return std::nullopt;
     }
 
-    if (tlb_index >= 0 && tlb_index < TLB_COUNT_1M) {
-        std::uint32_t size = 1 << 20;
-        return std::tuple(TLB_BASE_1M + size * tlb_index, size);
-    } else if (tlb_index >= 0 && tlb_index < TLB_COUNT_1M + TLB_COUNT_2M) {
-        auto tlb_offset = tlb_index - TLB_COUNT_1M;
+    // Only have 2MB TLBs for now
+    if (tlb_index >= 0 && tlb_index < TLB_BASE_2M) {
```

Comment:
A bit late to notice this, but I think this change is not quite right, should use TLB_COUNT_2M instead of TLB_BASE_2M?  Working in this area now, will try to fix/test.

Reply:
Thanks @kmabeeTT.
This function isn't used in BH which is why this didn't yield any issues. Once static TLBs are added, it would've been found.
Fixed it though